catalyst_voices/packages/catalyst_voices_services/lib/src/user/user_service.dart

```diff
@@ -0,0 +1,223 @@
+import 'dart:async';
+
+import 'package:catalyst_cardano_serialization/catalyst_cardano_serialization.dart';
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_services/catalyst_voices_services.dart';
+import 'package:logging/logging.dart';
+
+abstract interface class UserService {
+  Account? get account;
+
+  Stream<Account?> get watchAccount;
+
+  Keychain? get keychain;
+
+  Future<List<Keychain>> get keychains;
+
+  Stream<Keychain?> get watchKeychain;
+
+  Future<void> useActiveAccount();
+
+  Future<void> removeActiveAccount();
+
+  Future<void> switchToAccount(Account account);
+
+  Future<void> switchToKeychain(String id);
```

Comment:
If a keychain is related to the account, wouldn't it make sense to combine chaing both keychain & account?

Reply:
I tried to do it but didn't have any good ideas. There are few topics to keep in mind here:

1. `Account` is in `models` package while `Keychain` in `services`. We don't want `model` to depend on `services`.
2. `Account` can and should be fetched only after `Keychain` is unlocked. Otherwise we don't know anything about owner. Maybe there should be other abstraction describing this relation